src/app/network/DownloadManager.ts

```diff
@@ -185,4 +211,13 @@ export class DownloadManager {
       return newTask;
     }
   };
+
+  private static compareArraysByIdAndFolder = (arr1: DownloadItemType[], arr2: DownloadItemType[]) => {
+    if (arr1.length !== arr2.length) return false;
+
+    return arr1.every((obj1, index) => {
+      const obj2 = arr2[index];
+      return obj2?.id === obj1.id && obj2?.isFolder === obj1.isFolder;
+    });
+  };
```

Comment:
This function has same problem as before, if it has the same items in different order will not work correctly.
Also add tests for this function